File: Android/tuilivekit/livekit/src/main/java/com/trtc/uikit/livekit/component/beauty/basicbeauty/view/BeautyListAdapter.java
package com.trtc.uikit.livekit.component.beauty.basicbeauty.view;

import static com.trtc.uikit.livekit.component.beauty.basicbeauty.store.BasicBeautyState.ITEM_BEAUTY_CLOSE;
import static com.trtc.uikit.livekit.component.beauty.basicbeauty.store.BasicBeautyState.ITEM_BEAUTY_RUDDY;
import static com.trtc.uikit.livekit.component.beauty.basicbeauty.store.BasicBeautyState.ITEM_BEAUTY_SMOOTH;
import static com.trtc.uikit.livekit.component.beauty.basicbeauty.store.BasicBeautyState.ITEM_BEAUTY_WHITENESS;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.trtc.uikit.livekit.R;
import com.trtc.uikit.livekit.component.beauty.basicbeauty.store.BasicBeautyStore;
import com.trtc.uikit.livekit.component.beauty.basicbeauty.store.BasicBeautyState;

import java.util.ArrayList;
import java.util.List;

public class BeautyListAdapter extends RecyclerView.Adapter<BeautyListAdapter.ViewHolder> {
    private final Context          mContext;
    private final List<BeautyItem> mData                  = new ArrayList<>();
    private int              mCurrentBeautyPosition = -1;
    private BasicBeautyState mState;

    public BeautyListAdapter(Context context) {
        mContext = context;
        initData();
    }

    private void initData() {
        mState = BasicBeautyStore.getInstance().beautyState;

        mData.add(new BeautyItem(mContext.getString(R.string.common_beauty_item_close),
                R.drawable.beauty_item_close, ITEM_BEAUTY_CLOSE, null));
        mData.add(new BeautyItem(mContext.getString(R.string.common_beauty_item_smooth),
                R.drawable.beauty_item_smooth, ITEM_BEAUTY_SMOOTH, null));
        mData.add(new BeautyItem(mContext.getString(R.string.common_beauty_item_whiteness),
                R.drawable.beauty_item_whiteness, ITEM_BEAUTY_WHITENESS, null));
        mData.add(new BeautyItem(mContext.getString(R.string.common_beauty_item_ruddy),
                R.drawable.beauty_item_ruddy, ITEM_BEAUTY_RUDDY, null));
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.beauty_panel_item,
                parent, false);
        return new ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, @SuppressLint("RecyclerView") int position) {
        holder.mTextTitle.setText(mData.get(position).title);
        holder.mImageIcon.setImageResource(mData.get(position).icon);
        holder.mLayoutRoot.setTag(mData.get(position).type);
        if (mData.get(position).type == mState.mCurrentBeautyType.getValue()) {
            holder.mImageIcon.setBackgroundResource(R.drawable.beauty_item_select_background);
        } else {
            holder.mImageIcon.setBackgroundResource(R.drawable.beauty_item_not_select_background);
        }
        holder.mLayoutRoot.setOnClickListener(view -> {
            mState.mCurrentBeautyType.setValue(mData.get(position).type);
            notifyDataSetChanged();
        });
    }

    @Override
    public int getItemCount() {
        return mData.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public LinearLayout mLayoutRoot;
        public TextView     mTextTitle;
        public ImageView    mImageIcon;

        public ViewHolder(View itemView) {
            super(itemView);
            mLayoutRoot = itemView.findViewById(R.id.ll_root);
            mTextTitle = itemView.findViewById(R.id.tv_title);
            mImageIcon = itemView.findViewById(R.id.iv_icon);
        }
    }

    public static class BeautyItem {
        public String               title;
        public int                  icon;
        public int                  type;
        public View.OnClickListener listener;

        public BeautyItem(String title, int icon, int type, View.OnClickListener listener) {
            this.title = title;
            this.icon = icon;
            this.type = type;
            this.listener = listener;
        }
    }
}

File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/component/beauty/basicbeauty/view/BeautyListAdapter.kt
package com.trtc.uikit.livekit.component.beauty.basicbeauty.view

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.trtc.uikit.livekit.R

class BeautyListAdapter(
    context: Context
) : RecyclerView.Adapter<BeautyListAdapter.ViewHolder>() {
    private var currentBeautyType = BeautyType.NONE
    private var onItemClickListener: OnItemClickListener? = null

    private val data = listOf(
        BeautyItem(
            context.getString(R.string.common_beauty_item_close),
            R.drawable.beauty_item_close,
            BeautyType.CLOSE,
            null
        ),
        BeautyItem(
            context.getString(R.string.common_beauty_item_smooth),
            R.drawable.beauty_item_smooth,
            BeautyType.SMOOTH,
            null
        ),
        BeautyItem(
            context.getString(R.string.common_beauty_item_whiteness),
            R.drawable.beauty_item_whiteness,
            BeautyType.WHITENESS,
            null
        ),
        BeautyItem(
            context.getString(R.string.common_beauty_item_ruddy),
            R.drawable.beauty_item_ruddy,
            BeautyType.RUDDY,
            null
        )
    )

    fun setOnItemClickListener(listener: OnItemClickListener) {
        onItemClickListener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
        ViewHolder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.beauty_panel_item, parent, false)
        )

    @SuppressLint("RecyclerView", "NotifyDataSetChanged")
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = data[position]
        with(holder) {
            textTitle.text = item.title
            imageIcon.run {
                setImageResource(item.icon)
                setBackgroundResource(
                    if (item.type == currentBeautyType) {
                        R.drawable.beauty_item_select_background
                    } else {
                        R.drawable.beauty_item_not_select_background
                    }
                )
            }
            layoutRoot.apply {
                tag = item.type
                setOnClickListener {
                    currentBeautyType = item.type
                    onItemClickListener?.onItemClick(currentBeautyType)
                    notifyDataSetChanged()
                }
            }
        }
    }

    override fun getItemCount() = data.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val layoutRoot: LinearLayout = itemView.findViewById(R.id.ll_root)
        val textTitle: TextView = itemView.findViewById(R.id.tv_title)
        val imageIcon: ImageView = itemView.findViewById(R.id.iv_icon)
    }

    data class BeautyItem(
        val title: String,
        val icon: Int,
        val type: BeautyType,
        val listener: View.OnClickListener?
    )

    enum class BeautyType(val value: Int) {
        NONE(0),
        CLOSE(1),
        SMOOTH(2),
        WHITENESS(3),
        RUDDY(4)
    }

    fun interface OnItemClickListener {
        fun onItemClick(type: BeautyType)
    }
}
